web/core/components/issues/issue-detail-widgets/relations/quick-action-button.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+import React, { FC, useState } from "react";
+import { observer } from "mobx-react";
+import { Plus } from "lucide-react";
+import { ISearchIssueResponse, TIssueRelationTypes } from "@plane/types";
+import { CustomMenu, TOAST_TYPE, setToast } from "@plane/ui";
+// components
+import { ExistingIssuesListModal } from "@/components/core";
+// hooks
+import { useIssueDetail } from "@/hooks/store";
+// helper
+import { ISSUE_RELATION_OPTIONS } from "./helper";
+
+type Props = {
+  workspaceSlug: string;
+  projectId: string;
+  issueId: string;
+  customButton?: React.ReactNode;
+  disabled?: boolean;
+};
+
+export const RelationActionButton: FC<Props> = observer((props) => {
+  const { workspaceSlug, projectId, customButton, issueId, disabled = false } = props;
+  // state
+  const [relationKey, setRelationKey] = useState<TIssueRelationTypes | null>(null);
+  const { createRelation, isRelationModalOpen, toggleRelationModal } = useIssueDetail();
+
+  // handlers
+  const handleOnClick = (relationKey: TIssueRelationTypes) => {
+    setRelationKey(relationKey);
+    toggleRelationModal(issueId, relationKey);
+  };
+
+  // submit handler
+  const onSubmit = async (data: ISearchIssueResponse[]) => {
+    if (!relationKey) return;
+    if (data.length === 0) {
+      setToast({
+        type: TOAST_TYPE.ERROR,
+        title: "Error!",
+        message: "Please select at least one issue.",
+      });
+      return;
+    }
+
+    await createRelation(
+      workspaceSlug,
+      projectId,
+      issueId,
+      relationKey,
+      data.map((i) => i.id)
+    );
+
+    toggleRelationModal(null, null);
+  };
+
+  const handleOnClose = () => {
+    setRelationKey(null);
+    toggleRelationModal(null, null);
+  };
+
+  // button element
+  const customButtonElement = customButton ? <>{customButton}</> : <Plus className="h-4 w-4" />;
+
+  return (
+    <>
+      <CustomMenu customButton={customButtonElement} placement="bottom-start" disabled={disabled} closeOnSelect>
```

Comment:
Can we follow a similar pattern as the dropdown component here?

Reply:
A ticket has been raised for this [[WEB-1864]](https://app.plane.so/plane/projects/02c3e1d5-d7e2-401d-a773-45ecba45d745/issues/efe9a6a2-19b5-4766-967f-3d7e841fa34d) .I'll take care of it.